pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1267,19 +1270,35 @@ class Timedelta(_Timedelta):
                                  "(days,seconds....)")
 
             kwargs = {key: _to_py_int_float(kwargs[key]) for key in kwargs}
-
-            nano = convert_to_timedelta64(kwargs.pop('nanoseconds', 0), 'ns')
-            try:
-                value = nano + convert_to_timedelta64(timedelta(**kwargs),
-                                                      'ns')
-            except TypeError as e:
+            if not cls._req_any_kwargs_new.intersection(kwargs):
```

Comment:
This change means we now fail to raise on `pd.Timedelta(days=2, foo=9)`

Reply:
I am going to create a PR for this..